src/Ng/Program.cs

```diff
@@ -92,7 +93,7 @@ public static async Task MainAsync(string[] args)
                 job = NgJobFactory.GetJob(jobName, telemetryService, loggerFactory);
                 await job.RunAsync(arguments, cancellationTokenSource.Token);
             }
-            catch (ArgumentException ae)
+            catch (Exception ae) when (ae is ArgumentException || ae is KeyNotFoundException)
```

Comment:
In general, catching an `ArgumentException` here, and representing it as a "required argument id missing/invalid" is a bit of a stretch, since the `RunAsync` call is tried that runs complete job. `ArgumentException` can be thrown anywhere, not only at the argument validation level.
Ideally, we'd probably want a separate command line argument validation/consumption stage separate from running the job, but I'd imagine that would be quite out of scope for the intended change.

Reply:
Yup, I didn't want to fix the questionable design of the command line parsing -- I just wanted to make the current mess more usable.